### PR TITLE
Add av1e hierarchy level info.

### PR DESCRIPTION
**Pull request by RachelChengRJ**
This is a change to add hierarchy information for AV1 encoder.

### CODE REVIEW

**Review by Xuhenry1 — APPROVED** [review]
lgtm

File: va/va_enc_av1.h [hunk 1]
```diff
@@ -575,7 +578,12 @@ typedef struct  _VAEncPictureParameterBufferAV1
      */
     uint8_t     ref_frame_idx[7];
 
-    uint8_t     reserved8bits0;
+    /** \brief When hierachical_level_plus1 > 0, hierachical_level_plus1-1 indicates
+     *the current frame's level.when it > 0. B1, B2 setting in CodingType can be treated as B,
+     *hirachical level is determined by this variable.When hierachical_level_plus1  == 0,
+     *hierarchical level information still comes from coding_type.
+     */
+    uint8_t     hierarchical_level_plus1;
```
Comment (XinfengZhang) [inline va/va_enc_av1.h:586]: hi @RachelChengRJ could it be replaced by VAEncMiscParameterTemporalLayerStructure?

Comment (RachelChengRJ) [reply va/va_enc_av1.h:586]: Hi Carl, as we offline discussed, temporal layer info only provide  mini GOP structure inside whole sequence, not each frame’s layer info directly, so basically this change is fine, can be merged.

**Comment by RachelChengRJ** [comment]
> hi @RachelChengRJ , LGTM , but ,please help to rebase your code , and add more information in the commit message, dont forget the signoff. thanks

Thanks Carl for reminder, fixed. please help to review again.

**Review by XinfengZhang — APPROVED** [review]

File: va/va_enc_av1.h [hunk 2]
```diff
@@ -30,7 +30,7 @@
  *
  */
 
-#ifndef VA_ENC_AV1_H
+#ifndef VA_ENC_AV1_Hhier
```
Comment (XinfengZhang) [inline va/va_enc_av1.h:33]: ??? whats this, looks your s/ happened everywhere

**Review by XinfengZhang — CHANGES_REQUESTED** [review]

**Review by XinfengZhang — APPROVED** [review]

File: va/va_enc_av1.h (hunk 2 above)
[the remark this answers is not in the document]
Comment (RachelChengRJ) [reply va/va_enc_av1.h]: This is fixed, please help to check latest version.

**Pull request merged**